src/entities/database.js

```diff
@@ -284,6 +284,43 @@ class DB implements Database {
     )
   }
 
+  async queryLoosePendingTxs() {
+    const snapshotsTable = 'pending_snapshots'
+    const query = Q.sql.select().from(snapshotsTable)
+      .field('tx_hash')
+      .where('block_height = ?', Q.sql.select().from(snapshotsTable)
+        .field('MAX(block_height)'))
+      .union(Q.sql.select().from('txs')
+        .field('hash')
+        .where('tx_state = ?', TX_STATUS.TX_PENDING_STATUS)
+        .where('NOT EXISTS ?', Q.sql.select().from(snapshotsTable)
+          .field('1')
+          .where('tx_hash = hash')))
+      .toString()
+    const dbRes = await this.getConn().query(query)
+    this.#logger.debug('queryLoosePendingTxs:', query, dbRes)
+    return _.map(dbRes.rows, 'tx_hash')
+  }
+
+  async storeNewPendingSnapshot(block: Block) {
+    const loosePendingSet = await this.queryLoosePendingTxs()
+    const txHashes = _.map(block.txs, 'id')
+    const nextSnapshot = loosePendingSet.filter(hash => !txHashes.includes(hash))
```

Comment:
If we are using `lodash` anyways - there's a special function for that:

![image](https://user-images.githubusercontent.com/5585355/66322549-faf88500-e92a-11e9-971a-9e2abfc26c9e.png)

```
const nextSnapshot = _.difference(pendingSet, txHashes);
```